### PR TITLE
Adding tests for core/core/initutils and fixing potential bug

### PR DESCRIPTION
**Pull request by mohaidoss**
## PR Type:
Tests

___
## PR Description:
This PR includes the following changes:
- Added comprehensive unit tests for the functions in the `core/core/initutils.go` file.
- Fixed a potential bug in the `setSubmitBehavior` function in `core/core/initutils.go` by replacing `getter.GetKSCloudAPIConnector().GetCloudReportURL()` with `tenantConfig.GetCloudReportURL()`.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `core/core/initutils.go`: Replaced `getter.GetKSCloudAPIConnector().GetCloudReportURL()` with `tenantConfig.GetCloudReportURL()` in the `setSubmitBehavior` function to fix a potential bug.
- `core/core/initutils_test.go`: Added comprehensive unit tests for the functions in the `core/core/initutils.go` file. This includes tests for `getExceptionsGetter`, `policyIdentifierIdentities`, `getUIPrinter`, `GetSensorHandler`, `setSubmitBehavior`, `isScanTypeForSubmission`, `GetDefaultFrameworksPaths`, and `getDownloadReleasedPolicy`.
</details>

### CODE REVIEW

**Comment by mohaidoss** [comment]
Some tests are missing for functions that call `SetRegoObjects`, because I couldn't mock that function (AI seems clueless too)...  

If someone knowledgeable can help out, It would really help here! I'm new to the language :cry:

**Comment by mohaidoss** [comment]
@matthyx One of the tests are failing `TestGenerateTableWithCorrectHeadersAndRows`
Not expecting colors on the output..

**Comment by matthyx** [comment]
@mohaidoss can you rebase?

**Comment by mohaidoss** [comment]
@matthyx Yes, it is good now. I reverted my very recent changes

**Review by matthyx — CHANGES_REQUESTED** [review]

File: core/core/initutils_test.go [hunk 1]
```diff
@@ -240,3 +555,32 @@ func TestGetDefaultFrameworksPaths(t *testing.T) {
...
+	require.NoError(t, downloadReleasedPolicy.SetRegoObjects())
+
+	result := getDownloadReleasedPolicy(ctx, downloadReleasedPolicy)
+
+	assert.NotNil(t, result)
+}
+
+/* Should mock downloadReleasedPolicy.SetRegoObjects to return an error
+
+func TestGetDownloadReleasedPolicyError(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (mohaidoss) [reply core/core/initutils_test.go]: Removed

**Review by matthyx — APPROVED** [review]

**Pull request merged**